build/tools/SharedCodeGenerator/Outputs/MSBuildOutput.cs

```diff
@@ -0,0 +1,52 @@
+// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.Oryx.SharedCodeGenerator.Outputs
+{
+    [OutputType("msbuild")]
+    internal class MSBuildOutput : IOutputFile
+    {
+        private ConstantCollection _collection;
+        private string _directory;
+        private string _fileNamePrefix;
+
+        public void Initialize(ConstantCollection constantCollection, Dictionary<string, string> typeInfo)
+        {
+            _collection = constantCollection;
+            _directory = typeInfo["directory"];
+            _fileNamePrefix = typeInfo["file-name-prefix"];
+        }
+
+        public string GetPath()
+        {
+            var name = _collection.Name.Camelize();
+            name = char.ToLowerInvariant(name[0]) + name.Substring(1);
+            return Path.Combine(_directory, _fileNamePrefix + name + ".props");
+        }
+
+        public string GetContent()
+        {
+            StringBuilder body = new StringBuilder();
+            var autoGeneratedMessage = Program.BuildAutogenDisclaimer(_collection.SourcePath);
+            body.AppendLine($"<!-- {autoGeneratedMessage} -->"); // Can't use AppendLine becuase it appends \r\n
```

Comment:
i feel like this method is more like setcontent

Reply:
I did not get what you mean. This is a method on stringbuilder

Reply:
This method is part of `IOutputFile` but irrespective of that this api is only for getting the content which is finally going to be written to a file that is generated at the call site. So `GetContent` feels correct to me.